File: src/main/kotlin/com/wynnlab/spells/mage/Heal.kt
package com.wynnlab.spells.mage

import com.wynnlab.spells.SpellL
import com.wynnlab.spells.SpellData
import org.bukkit.Bukkit
import org.bukkit.Particle
import org.bukkit.Sound
import org.bukkit.attribute.Attribute
import org.bukkit.entity.Player
import org.bukkit.event.entity.EntityRegainHealthEvent

class Heal(player: Player) : SpellL(player, 41, SpellData.HEAL) {
    override fun tick() {
        if (tick % 20 == 0) {
            player.spawnParticle(Particle.PORTAL, player.location.clone().add(0.0, 0.5, 0.0), 144, 4.0, 0.0, 4.0, 0.1)
            player.spawnParticle(Particle.CRIT_MAGIC, player.location.clone().add(0.0, 0.3, 0.0), 144, 4.0, 0.0, 4.0, 0.1)
            player.spawnParticle(Particle.FIREWORKS_SPARK, player.location.clone().add(0.0, 1.0, 0.0), 16, 0.3, 1.0, 0.3, 0.05)
            player.playSound(player.location, Sound.ENTITY_EVOKER_CAST_SPELL, 0.5f, 1.5f)
            player.playSound(player.location, Sound.BLOCK_LAVA_EXTINGUISH, 1f, 1f)
            player.health = (player.health + 50).coerceAtMost(player.getAttribute(Attribute.GENERIC_MAX_HEALTH)!!.value) // Heal Health
            for (e in player.getNearbyEntities(4.0, 4.0, 4.0)) {
                if (e is Player) {
                    e.health = (e.health + 50).coerceAtMost(e.getAttribute(Attribute.GENERIC_MAX_HEALTH)!!.value) // Heal Health
                    Bukkit.getPluginManager().callEvent(
                        EntityRegainHealthEvent(player, 50.0, EntityRegainHealthEvent.RegainReason.CUSTOM) // Heal Health
                    )
                    player.spawnParticle(Particle.FIREWORKS_SPARK, e.location.clone().add(0.0, 1.0, 0.0), 16, 0.3, 1.0, 0.3, 0.05)
                }
            }
        }
    }
}

File: src/main/resources/classes/mage/heal.py
from org.bukkit import Bukkit, Particle, Sound
from org.bukkit.attribute import Attribute
from org.bukkit.entity import Player
from org.bukkit.event.entity import EntityRegainHealthEvent

from com.wynnlab.spells import PySpell

class Spell(PySpell):
    def tick(self):
        if self.t % 20 > 0:
            return

        self.particle(self.player.getLocation().clone().add(0, .5, 0), Particle.PORTAL, 144, 4, 0, 4, .1)
        self.particle(self.player.getLocation().clone().add(0, .3, 0), Particle.CRIT_MAGIC, 144, 4, 0, 4, .1)
        self.particle(self.player.getLocation().clone().add(0, 1, 0), Particle.FIREWORKS_SPARK, 16, .3, 1, .3, .05)
        self.sound(Sound.ENTITY_EVOKER_CAST_SPELL, .5, 1.5)
        self.sound(Sound.BLOCK_LAVA_EXTINGUISH, 1, 1)

        amount = self.player.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue() // 10
        PySpell.heal(self.player, amount)
        #self.player.sendMessage('{0}{1}{2}'.format(PySpell.colorText('[', '4'), PySpell.colorText('+{a}'.format(a=amount), 'c'), PySpell.colorText(']', '4')))

        for p in self.player.getNearbyEntities(4, 4, 4):
            if not isinstance(p, Player):
                continue

            PySpell.heal(p, amount)
            #p.sendMessage('{0}{1}{2}{3}'.format(PySpell.colorText('[', '4'), PySpell.colorText('+{a}'.format(a=amount), 'c'), PySpell.colorText(']', '4'), PySpell.colorText('({pl})'.format(self.player.getName()), '7')))
            Bukkit.getPluginManager().callEvent(EntityRegainHealthEvent(p, 50, EntityRegainHealthEvent.RegainReason.CUSTOM))

            self.particle(p.getLocation().clone().add(0, 1, 0), Particle.FIREWORKS_SPARK, 16, .3, 1, .3, .05)
